[improvement-#11879]Keyword Search Improvement

fuchanghai opened the pull request:
## Purpose of the pull request

add a shortcut Key


## Brief change log

add a shortcut Key


## Verify this pull request



This pull request is code cleanup without any test coverage.

 this close #11879

zhongjiajie committed:
[fix][test] Fix flaky test in CI (#12017)

stalary committed:
[Improvement][Test] Remove Powermock in dolphinscheduler-task-plugin (#12153)

* Remove the usage of powermock in task-plugin module

TyrantLucifer committed:
[Feature][doc] Add a guide to nominate new committer (#12175)

Co-authored-by: Kirs <ACM_China@yeah.net>

jieguangzhou committed:
fix ml group (#12172)

darrkz committed:
[Document][Improvement] Polish docs for standalone deployment (#12181)

* Polish docs for standalone deployment

EricGao888 committed:
[Improvement][Test] Remove powermock in service and server modules (#12164)

* Remove powermock and refactor some related code in dolphinscheduler-service and dolphinscheduler-server modules
* Remove redundant comments
* Add null check

zhongjiajie committed:
[feat][ci] Add pre-commit for easy spotless config (#12192)

* Use pre-commit to handle spotless check
* Migrate exists python pre-commit-config into dir root one
* Change related document

fuchanghai committed:
[improvement] improve relation between notification strategy and alarm group on ui (#12200)

* [improvement] improve relation between notification strategy and alarm group

* [improvement] change to use ```&&```

Co-authored-by: Jackie <‘2875334588@qq.com’>

rickchengx committed:
[Improvement-12144] Remove powermock in dolphinschduler-api (#12169)

fuchanghai committed:
[fix-#11753] send alert error alert data id (#11774)

* [fix-#11753] send alert error alert data id

Co-authored-by: fuchanghai <changhai.fu@marketingforce.com>

guowei-su committed:
[DS-12154][worker] Optimize the log printing of the worker module (#12183)

* [DS-12154][worker] Optimize the log printing of the worker module according to the log specification.

HomminLee committed:
[Improvement-11678][API] Improvement the error message when batch delete workflow (#11682)

* [Improvement] Improvement the error message when batch delete workflow

rickchengx committed:
[Improvement-11880][Hivecli] Improve the UI of HiveCli (#11882)

jieguangzhou committed:
[Feature][TASK-PLUGIN] AWS Database Migration Service (DMS) (#11868)

* add aws dms

caishunfeng committed:
[Release][Doc] add 3.1.0 release (#12227)

* [Release] add 3.1.0 release to doc

ruanwenjun committed:
Remove workerGroup in registry (#12217)

Amy0104 committed:
[Improvement][UI] Set the `required` to be reactive in the task modal. (#12225)

caishunfeng committed:
[Improvement] remove log-server and server module (#12206)

* remove log-server module

* remove dolphinscheduler-server module

EricGao888 committed:
[Improvement][Test] Remove the usage of powermock in datasource-plugin module (#12194)

* Remove the usage of powermock in datasource-plugin module

* Switch datasource-plugin module to use PropertyUtils in common module

tracehh committed:
[Improvement][Master] LowerWeightRoundRobin doSelect may be null (#12159)

ruanwenjun committed:
Fix hive datasource connection leak (#12226)

DarkAssassinator committed:
[fix#12195] Change date to timestamp to support cross diff time zones (#12239)

* fix DS support cross time zone, use timestamp replace date

zhongjiajie committed:
[fix][doc] Correct release export key step (#12228)

We should only add KEY to release/KEYS instead of
dev/KEYS, according to
https://lists.apache.org/thread/2gppxc04n0r87vq4qy4j3n52zwhqd442

DarkAssassinator committed:
just correct wrong word (#12174)

DarkAssassinator committed:
just add WorkerGroupMapper UT (#12242)

DarkAssassinator committed:
[Doc] Add chmod command after unzip DS tar.gz (#12210)

FlechazoW committed:
[Improvement][Task Plugin] Modify the comment of 'deployMode'. (#12163)

aiwenmo committed:
[Fix-12254] [Registry] Modify the type of field named data into text (#12256)

Co-authored-by: wenmo <32723967+wenmo@users.noreply.github.com>

davidzollo committed:
[doc] Update slack invitation link (#12258)

lynn-builds committed:
[Feat][UI] Set up UI setting component in profile page (#11226)

* Set up UI setting component in profile page

* Add UI setting page to the route, remove UI setting section from the profile

* Add log auto refresh timer store

* Add logTimer to pinia store and cookies

* Read logTimer from pinia store, and delay getLogs by passing the logTimer to setInterval in refreshLogs function

* Add log timer to getLogs

* Fine tune view log, add auto refresh based on time interval set in UI setting

* Add useI18n to ui setting

* Set up UI setting component in profile page

* Add UI setting page to the route, remove UI setting section from the profile

* Add log auto refresh timer store

* Add logTimer to pinia store and cookies

* Read logTimer from pinia store, and delay getLogs by passing the logTimer to setInterval in refreshLogs function

* Add log timer to getLogs

* Fine tune view log, add auto refresh based on time interval set in UI setting

* Add useI18n to ui setting

* [Feat][UI] Add UI setting page in the project.

* [Feat][UI] Add UI setting page in the project.

* Remove logTimer in cookies

* Clear timer id, set VITE_APP_DEV_WEB_URL back to default

* Clear time id in dag

* [Feat][UI] Add license header.

* [Feat][UI] Remove console.

* [Fix][UI] Fix log timer types.

Co-authored-by: songjianet <1778651752@qq.com>

TyrantLucifer committed:
[Hotfix][Docs] Fix 404 dead links (#12261)

fuchanghai committed:
[improvement-#11879]Keyword Search Improvement

fuchanghai commented:
@Amy0104

fuchanghai commented:
@Amy0104 @songjianet  i need your approve

ruanwenjun committed:
Fix mysql registry doesn't support array (#12255)

Tianqi-Dotes committed:
[Feature][task-plugins] aws datasync task plugins (#11865)

* submit datasync task

* add aws software package license

Amy0104 reviewed (commented):
It seems to be better to wrap a common search component.

EricGao888 committed:
Remove powermock from DatasyncTaskTest (#12270)

guowei-su committed:
[DS-12131][master] Optimize the log printing of the master module acc… (#12152)

* [DS-12131][master] Optimize the log printing of the master module according to the log specification.

Tianqi-Dotes committed:
delete useless exclusions (#12273)

delete software.amazon.awssdk datasync useless dependency exclusions

calvinjiang committed:
fix this issue (#12281)

fuchanghai committed:
[doc] Correct descriptions in glossary.md (#12282)

fuchanghai committed:
[improve] Correct some log in scheduler (#12278)

EricGao888 committed:
[Improvement][Test] Fully remove the usage of powermock from the whole project (#12244)

* Fully remove the usage of powermock from the whole project

* Upgrade org.reflections to 0.10.12

Chris-playground committed:
create oracle help.md (#12218)

* create oracle help.md

* Add files via upload

allow Oracle to choose schema

* move pic under img and add title for oracle.md

wangfann committed:
[Fix#12231][scheduler-plugin] fix the bug which some scheduled tasks are not triggered on time (#12233)

Co-authored-by: fan.wang <fan.wang@jiduauto.com>

tuchg committed:
Update ZooKeeper minimum version requirements (#12284) (#12288)

zhongjiajie committed:
[ci] Milestone and Label are required before the PR merged (#12222)

To make our release process easier, I plan to mark milestone
and label is required before the PR merged, they use for

* label: In this path we required label with at least on with
  'feature|bug|improvement|document|chore' to make our changelog
  more convenient
* milestone: For easier to cherry-pick PR, currently some of
  our milestone only exists in issue and not exists in related
  PR, it may

xdu-chenrj committed:
[python] Add new resource plugin: gitlab, OSS and S3 (#12025)

Co-authored-by: Jiajie Zhong <zhongjiajie955@gmail.com>

wangfann committed:
[ fix: #11590 ][upgrade] Fix the bugs when I upgrade ds from v1.3.9 to v3.0.0 ,such as file cannot be found or column not found etc (#11619)

* fix the bugs which upgrade ds metabase from v1.3.9 to v3.0.0 with mysql

* [Fix-11590][upgrade] Fix the bugs when I upgrade ds from v1.3.9 to v3.0.0 ,such as file cannot be found or column not found etc

* fix: #11590 add process_instance_code_state_index with table t_ds_process_instance to improve query efficiency for the method of findLastProcessInterval

* Fix the bug  that cannot find the files when upgrade ds version with postgresql

Co-authored-by: fan.wang <fan.wang@jiduauto.com>

jackfanwan committed:
Set the id to null when copying t_ds_schedules (#12280)

Co-authored-by: fanwanlong <fanwanlong@kezaihui.com>

EricGao888 committed:
[Improvement][Test] Migrate all UT cases from jUnit4 to jUnit5 in task-plugin module as an example (#12299)

* Migrate all UT cases from jUnit4 to jUnit5 in task-plugin module as an example

rickchengx committed:
[Improvement-12296][Test] Close static mock (#12300)

boy-xiaozhang committed:
[fix] Fix error problem on h2 startup data quality rule management page (#12108)

Co-authored-by: jegger <zhangbq3306@163.com>

Commit:
[improvement] add a search component

zhongjiajie committed:
[dev] Easier release: cherry-pick, changelog, contributor (#11478)

Add script for  easier release including cherry pick,
generating changelog and contributor list

* Auto cherry-pick: `python release.py cherry-pick`
* Generate changelog: `python release.py changelog`
* Generate contributor: `python release.py contributor`

close: #11289
related: #12222

zhongjiajie committed:
[doc] Change release process base on new tool (#12324)

ref: #11478 we already have a new tool for releasing

rickchengx committed:
[Improvement-12289][k8s] update the common.properties in values.yaml (#12290)

Amy0104 committed:
[Feature][UI] Revise the  `Search` component.

Commit:
[improvement] add search components
